### DISCUSSION TITLE
How to approach multi-devices development? (beginner)

### DISCUSSION THREAD

**Discussion by wsw70** [quasarframework/quasar#9647] [General - Components / Directives / etc]
I have a few years of "home" development experience and I use Quasar for a year or so, for apps that run on a web browser on a PC screen.

I now would like to adapt it to my phone (and ideally, to other devices - but at least to my phone) by reshuffling the components.

**Main question**: is there a "magical" way where the reshuffling is automated so that the components (and font sizes etc.)  "best adapt" to the device?

Since I guess that the answer is "no", is the proper approach the one mentioned in the [platform detection documentation](https://next.quasar.dev/options/platform-detection) to have a construction such as

```
<div v-if="$q.platform.is.mobile">
  here are all the components in the configuration for mobiles
</div>
<div v-else>
  here are, again, all the components - this time in the configuration for the web site
</div>
```

In other words: **do I have to build two configurations and use in each of them the components twice?** (except if they are components that are fine in both cases, but it may be easier to duplicate them to avoid having plenty of `v-if` all over the place)

**Comment by ontwikkelfabriek** [comment]
@wsw70 
> do I have to build two configurations and use in each of them the components twice?

You can but you certainly don't have to. You can have the same **responsive** layout/components(using flexbox) used in native mobile build(ios and android, using cordova), mobile browsers and for desktop browsers.

**Comment by turigeza** [comment]
Just a quick note on `$q.platform.is.mobile` is also `true` to super high resolution big tablets. So it's not a good way to create a responsive layout.
